### ISSUE TITLE
Space added for raw SQL?

### ISSUE THREAD

**Issue by arichiardi** [layerware/hugsql#41]
I have a weird case where I wanted to add a raw "left or "right" to my sql, as it is:

```
UPDATE workflows
SET scan_:sql:side = :scan
WHERE id = :id
RETURNING *
```

Unfortunately this does not seem to work as the computed sqlvec is:
`["UPDATE workflows\nSET scan_ right = ?\nWHERE id = ?\nRETURNING *" "z.\\+g" 6N]`

(The garbage comes from a generator, but there is a space between `scan_` and `right`)

Am I taking the right approach?

**Comment by csummers** [comment]
This may be bug.  I know there were a few edge cases with whether or not to add a space for clojure expressions if the returned string did not account for any space.  Raw SQL parameters may need to have  different behavior for the intuitive use.  I'll look into this.

The work-around for your particular issue is to pass in the entire column name "scan_left" or "scan_right".  (And, once you're passing in the entire column name, you might as well use the `:i` identifier param type.)

**Comment by arichiardi** [comment]
Uhm, I used:
```
UPDATE workflows
/*~ (if (= "left" (:side params)) */
SET scan_left = :scan
/*~*/
SET scan_right = :scan
/*~ ) ~*/
WHERE id = :id
RETURNING *
```

It works fine...

**Comment by csummers** [comment]
That's great, too.  Thanks for the bug report!

**Comment by csummers** [comment]
This should be in the next release (0.5.0) later this week.  Thanks for the report!

This bug prompted me to reconsider some earlier decisions on whether or not to provide a "spacer" when a parameter or expression returns a value that might need one in the context of the SQL.  I've now decided to trim less whitespace and newlines when parsing and skip adding a spacer altogether.  There are a few edge cases where this change might break someone's existing code, hence the version bump to 0.5.0, but I think in the longer term this is the right thing to do.

**Comment by arichiardi** [comment]
Agree with you and can't wait for the next release! 😃

**Comment by csummers** [comment]
This is released in 0.4.8 (finally!).  Thanks!
